File: library/src/main/java/me/shkschneider/skeleton/network/BytesRequest.java
package me.shkschneider.skeleton.network;

import android.support.annotation.NonNull;

import com.android.volley.NetworkResponse;
import com.android.volley.Response;
import com.android.volley.toolbox.HttpHeaderParser;
import com.android.volley.toolbox.JsonRequest;

import me.shkschneider.skeleton.data.MimeTypeHelper;

// <https://github.com/DWorkS/VolleyPlus>
public class BytesRequest extends JsonRequest<byte[]> {

    public BytesRequest(final int method, @NonNull final String url, @NonNull final String requestBody, final Response.Listener<byte[]> listener, final Response.ErrorListener errorListener) {
        super(method, url, requestBody, listener, errorListener);
    }

    public BytesRequest(@NonNull final String url, final Response.Listener<byte[]> listener, final Response.ErrorListener errorListener) {
        super(Method.GET, url, null, listener, errorListener);
    }

    public BytesRequest(final int method, @NonNull final String url, final Response.Listener<byte[]> listener, final Response.ErrorListener errorListener) {
        super(method, url, null, listener, errorListener);
    }

    @Override
    protected Response<byte[]> parseNetworkResponse(@NonNull final NetworkResponse response) {
        return Response.success(response.data, HttpHeaderParser.parseCacheHeaders(response));
    }

    @Override
    public String getBodyContentType() {
        return MimeTypeHelper.APPLICATION_OCTETSTREAM;
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/network/BytesRequest.kt
package me.shkschneider.skeleton.network

import com.android.volley.NetworkResponse
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.HttpHeaderParser
import com.android.volley.toolbox.JsonRequest

import me.shkschneider.skeleton.data.MimeTypeHelper

// <https://github.com/DWorkS/VolleyPlus>
class BytesRequest : JsonRequest<ByteArray> {

    constructor(method: Int, url: String, requestBody: String, listener: Response.Listener<ByteArray>, errorListener: Response.ErrorListener) : super(method, url, requestBody, listener, errorListener)

    constructor(url: String, listener: Response.Listener<ByteArray>, errorListener: Response.ErrorListener) : super(Request.Method.GET, url, null, listener, errorListener)

    constructor(method: Int, url: String, listener: Response.Listener<ByteArray>, errorListener: Response.ErrorListener) : super(method, url, null, listener, errorListener)

    override fun parseNetworkResponse(response: NetworkResponse): Response<ByteArray> {
        return Response.success(response.data, HttpHeaderParser.parseCacheHeaders(response))
    }

    override fun getBodyContentType(): String {
        return MimeTypeHelper.APPLICATION_OCTETSTREAM
    }

}
